java/adapter/orc/src/main/java/org/apache/arrow/adapter/orc/OrcReader.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.adapter.orc;
+
+import java.io.IOException;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.ipc.ArrowReader;
+
+/**
+ *  Orc Reader that allow accessing orc stripes in Orc file.
+ *  This orc reader basically acts like an ArrowReader iterator that
+ *  iterate over orc stripes. Each stripe will be accessed via an
+ *  ArrowReader.
+ */
+public class OrcReader implements AutoCloseable {
+  private final OrcReaderJniWrapper jniWrapper;
+  private BufferAllocator allocator;
+
+  private final long id;
```

Comment:
please document what ID is.  there is probably a better name.  Maybe orc_pointer?